### PR TITLE
Sync lints and fix offenders

### PR DESCRIPTION
**Pull request by goderbauer**
We have lots of great lints enabled in the main flutter repository and the `analysis_options.yaml` file in this package repository has not kept up. This PR enabled all the lints from the main repository and fixes all offenders.

### CODE REVIEW

File: packages/extension_google_sign_in_as_googleapis_auth/example/lib/main.dart [hunk 1]
```diff
@@ -28,6 +28,9 @@ void main() {
 
 /// The main widget of this demo.
 class SignInDemo extends StatefulWidget {
+  /// Creates the main widget og this demo.
```
Comment (a14n) [inline packages/extension_google_sign_in_as_googleapis_auth/example/lib/main.dart:31]: typo: of

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: packages/pigeon/test/version_test.dart [hunk 2]
```diff
@@ -11,7 +11,7 @@ void main() {
   test('pigeon version matches pubspec', () {
     final String pubspecPath = '${Directory.current.path}/pubspec.yaml';
     final String pubspec = File(pubspecPath).readAsStringSync();
-    final RegExp regex = RegExp('version:\s*(.*?) #');
+    final RegExp regex = RegExp('version:s*(.*?) #');
```
Comment (a14n) [inline packages/pigeon/test/version_test.dart:14]: Shouldn't it be a raw String to keep `\s` ? `RegExp(r'version:\s*(.*?) #');`

**Review by a14n — APPROVED** [review]
LGTM

**Comment by ditman** [comment]
Thanks for these fixes! We're doing something similar in flutter/plugins, to eventually use the same analysis_options.yaml as flutter/flutter: https://github.com/flutter/plugins/pull/3703

**Pull request merged**